File: src/main/java/com/extendedae_plus/mixin/impl/bridge/HelperProviderButtons.java
package com.extendedae_plus.mixin.impl.bridge;

import com.extendedae_plus.client.render.widgets.button.EAEPActionButton;

import java.util.List;

/// 用于更新按钮布局, 在 `Provider & Interface` 的 `updateBeforeRender` 中调用
public interface HelperProviderButtons {
    void eaep$updateButtonsStates();

    List<EAEPActionButton> eaep$getScalingButtons();
}

File: src/main/kotlin/com/extendedae_plus/mixin/helper/HelperProviderButtons.kt
package com.extendedae_plus.mixin.helper

import com.extendedae_plus.client.render.widgets.button.EAEPButton

/** 用于更新按钮布局, 在 `Provider & Interface` 的 `updateBeforeRender` 中调用 */
interface HelperProviderButtons {
    fun `eaep$updateButtonsStates`()

    fun `eaep$getButtons`(): MutableList<out EAEPButton>
}
